src/lib/factories.js

```diff
@@ -81,12 +82,13 @@ export function createShorthand(Component, mapValueToProps, val, options = {}) {
   // Use key, childKey, or generate key
   if (_.isNil(props.key)) {
     const { childKey } = props
+    const { generateKey = true } = options
 
     if (!_.isNil(childKey)) {
       // apply and consume the childKey
       props.key = typeof childKey === 'function' ? childKey(props) : childKey
       delete props.childKey
-    } else if (valIsString || valIsNumber) {
+    } else if (generateKey && (valIsString || valIsNumber)) {
```

Comment:
Currently, setting `generateKey: false` can still result in creating a key if `childKey` is passed.  I think this check can be hoisted up to line 83.  If `generateKey` is false, I would expect the factory to not perform any operation related to keys. What are your thoughts?

Reply:
This is how the check was made in the previous implementation (before it was removed). I guess the question is: is there value in being able to disable an explicitly specified `childKey` in this way? I believe you can already achieve this by passing in `overrideProps: { childKey: null }`. There's also the discrepancy with `key`, which bypasses the conditional.

If the logic stays as is, the option could be renamed to `autoGenerateKey` or `allowAutoKey` (etc.) to signify the intent better. Alternatively the conditional logic could be rejigged so that passing `overrideProps: { key: null }` achieves the same thing, removing the need for the extra option altogether, but it would require us to differentiate between `null` and `undefined` (potentially brittle?).

Reply:
I see your point.  The shortcoming of using `overrideProps` is that it would override the user's key also, even when they explicitly define one.  I do like `autoGenerateKey`.  This way, it can be set to false and users can still choose to pass a key or not and get intuitive results.

Reply:
> The shortcoming of using `overrideProps` is that it would override the user's key also, even when they explicitly define one.

Ah I overlooked that somehow, however there's still `defaultProps` which could work:

```js
...
} else if (props.key === undefined && (valIsString || valIsNumber)) {
...
```

```js
// At call time:
ItemHeader.create(header, { defaultProps: { key: null } })
```

The obvious downside is that the semantics of that are far less clear compared to a for-purpose boolean flag (and it's a bit verbose).